api/src/services/config-watcher.js

```diff
@@ -135,6 +139,13 @@ const load = () => {
 };
 
 const listen = () => {
+  fs.watch(environment.webappPath, () => {
+    debouncedUpdateServiceWorker();
+  });
+  fs.watch(environment.loginPath, () => {
+    debouncedUpdateServiceWorker();
+  });
+
```

Comment:
This is only for dev mode, right? The files in prod should never change without a container restart?

If so, then I don't think we should watch in prod for performance reasons. We could either...

1. Wrap this in a big old `!isProduction` check which is easy but crude because it means dev-only code is still shipping to prod even if it's not executing (eg: unused lodash and fs dependencies).
2. implement the watcher in Grunt or some other build tool, which is great for separation from prod code, but we'd still need to do something in api to actually trigger the build of the SW (add an update SW api, or delete the file and check exists when serving, or call the generate service worker service from grunt, or... idk).

What do you think?

Reply:
Yea, this was on my questions list. 
I don't particularly like 1, which is arguably the easiest. 
I think an update SW API could work well, but it will complicate grunt a bit (grunt never actually calls API now). I definitely don't want to watch docs to trigger rebuilds :D

Reply:
I wondered about sending a process signal to API to trigger it so it's not exposed publicly in production but we don't do that anywhere and it means it has to be a local process.

I think the HTTP endpoint is fine with admin auth check as long as finding API isn't too hard.

Reply:
Done, added an endpoint!